### PR TITLE
[feat] Support CUDA Fortran file extensions in SingleSource

### PR DESCRIPTION
**Pull request by teojgo**
* Support the '.cuf' and '.CUF' file extensions when guessing the
  programming language in the `SingleSource` build system.

* Use the CUDA Fortran extension in corresponding regression test.

### CODE REVIEW

File: cscs-checks/prgenv/cuda-fortran/cuda_fortran_check.py [hunk 1]
```diff
@@ -24,13 +25,15 @@ def __init__(self):
...
-            self.build_system.fflags = [
-                'CUDA_HOME=$CUDATOOLKIT_HOME',
-                '-ta=tesla:cc60', '-Mcuda=cuda10.2'
+        if cdt == '20.08':
+            self.build_system.fflags += [
+                'CUDA_HOME=$CUDATOOLKIT_HOME', '-Mcuda=cuda10.2'
             ]
+        else:
+            # FIXME: workaround when CUDA 11.0 is the default version
+            self.modules += ['cudatoolkit/10.2.89_3.29-7.0.2.1_3.5__g67354b4']
```
Comment (vkarak) [inline cscs-checks/prgenv/cuda-fortran/cuda_fortran_check.py:39]: Why does this test require CUDA 10?

Comment (teojgo) [reply cscs-checks/prgenv/cuda-fortran/cuda_fortran_check.py:39]: I am adding a better description.

Comment (jgphpc) [reply cscs-checks/prgenv/cuda-fortran/cuda_fortran_check.py:39]: pgi/20.x does not support cuda/11

**Review by vkarak — APPROVED** [review]
lgtm

**Comment by vkarak** [comment]
@jenkins-cscs retry daint

**Pull request merged**